packages/framework/src/Controller/Admin/DefaultController.php

```diff
@@ -160,6 +166,48 @@ protected function addWarningMessagesOnDashboard(): void
                 ],
             );
         }
+
+        $notSetTermsAndConditionsArticles = $this->legalConditionsFacade->findNullSettingValues(Setting::TERMS_AND_CONDITIONS_ARTICLE_ID);
+
+        foreach ($notSetTermsAndConditionsArticles as $notSetTermsAndConditions) {
+            $domain = $this->domain->getDomainConfigById($notSetTermsAndConditions->getDomainId());
+
+            $this->addErrorFlashTwig(
+                t('<a href="{{ url }}">Terms and conditions article for domain {{ domainName}} is not set.</a>'),
+                [
+                    'url' => $this->generateUrl('admin_legalconditions_termsandconditions'),
+                    'domainName' => $domain->getName(),
+                ],
+            );
+        }
```

Comment:
What about 

```
foreach ($this->domain->getAll() as $domainConfig) {
    if ($this->setting->getForDomain(Setting::TERMS_AND_CONDITIONS_ARTICLE_ID, $domainConfig->getId()) === null) {
        $this->addErrorFlashTwig(
            t('<a href="{{ url }}">Terms and conditions article for domain {{ domainName}} is not set.</a>'),
            [
                'url' => $this->generateUrl('admin_legalconditions_termsandconditions'),
                'domainName' => $domainConfig->getName(),
            ],
        );
    }
    
     // if for privacy policy
     
     // if for cookie article
}
```

That way you don't need to create a new method in LegalConditionsFacade (and you don't need to autowire it into this controller), SettingValueRepository may remain unchanged. 
As a bonus, you save a few DB queries as the setting values are loaded at once (and at this point are already loaded).

Reply:
Oh i see, that is much more elegant approach. Thank you, rewrote it as such.